receiver.go

```diff
@@ -569,20 +579,39 @@ func (r *Receiver) mux() {
 		})
 	}()
 
+	var (
+		// counter that accumulates the settled delivery count.
+		// once the threshold has been reached, the counter is
+		// reset and a flow frame is sent.
+		settlementCount uint32
+
+		// contains the number of settlements that trigger a flow frame.
+		settlementThreshold uint32
+	)
+
+	if r.autoSendFlow {
+		settlementThreshold = r.l.linkCredit
+		if settlementThreshold > 1 {
```

Comment:
I'm going to rework this math a bit as I'm not happy with it and I think it's broken for some pathological cases.